core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -492,12 +491,15 @@ public void clearDataMaps(AbsoluteTableIdentifier identifier, boolean launchJob)
         tableIndices = allDataMaps.get(tableUniqueName);
       }
     }
-    if (null != carbonTable && tableIndices != null && launchJob) {
-      try {
-        DataMapUtil.executeDataMapJobForClearingDataMaps(carbonTable);
-      } catch (IOException e) {
-        LOGGER.error("clear dataMap job failed", e);
-        // ignoring the exception
+    if (launchJob) {
+      CarbonTable carbonTable = getCarbonTable(identifier);
+      if (null != carbonTable && tableIndices != null) {
```

Comment:
you can also judge tableIndices != null , then getCarbonTable, such as:
if (launchJob && tableIndices != null)

Reply:
done